### PR TITLE
Node.js: Adding health check endpoints for a Kubernetes environment

### PR DESCRIPTION
**Pull request by pacostas**
## Summary

This PR adds the `/actuator/health` health check endpoint on the Node.js applications.

* This route is necessary in a Kubernetes environment to determine whether an application's status is healthy or not.
* This route exists on java applications, by adding them to the Node.js applications, we keep consistency across the samples.

This PR adds also a logging function for  the Node.js apps to log any incoming requests, which is helpful to know whether the request has been received by the server.

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: nodejs/no-package-manager/server.js [hunk 1]
```diff
@@ -2,16 +2,38 @@ const http = require('http')
...
-}
+<head>
+  <title>Powered By Paketo Buildpacks</title>
+</head>
+<body>
+  <img style="display: block; margin-left: auto; margin-right: auto; width: 50%;" src="https://paketo.io/images/paketo-logo-full-color.png"></img>
+</body>
+</html>`);
+  } else if (
+    request.url === "/actuator/health/readiness" &&
```
Comment (mhdawson) [inline nodejs/no-package-manager/server.js:21]: Just wondering how you chose those endpoints. I was looking at https://nodeshift.dev/nodejs-reference-architecture/operations/healthchecks/#endpoints and from that we might use /readyz and /livez

Comment (mhdawson) [reply nodejs/no-package-manager/server.js:21]: I guess maybe it is to match the Java examples?

Comment (pacostas) [reply nodejs/no-package-manager/server.js:21]: Yes, to match the routes from the java example

**Comment by pacostas** [comment]
> I'm curious why you need these specific paths exposed in the samples. Can you expand on that?
> 
> The samples already have health endpoints. They're also pretty simple so liveness/readiness are really the same thing. Adding more paths and options makes the sample apps more complicated. Just trying to weigh the pros/cons here. Thanks

The node.js examples do not have a readiness or a liveness endpoints.

**Comment by mhdawson** [comment]
At one point I thought that some of the examples (the Java ones) already had health endpoints but that turns out not to be the case. 

In that context I don't have a strong feeling one way or the other in terms of adding them.  It could make it easier/better to use the samples in the context of deployment to Kubernetes environments, but its pretty easy to copy the examples when that is needed.

**Comment by mhdawson** [comment]
I will add though, that if we do want to show configuring the endpoints then I don't think we should be pointing them to the app endpoint and we'd want to use different endpoints for readiness and liveness.

**Comment by pacostas** [comment]
I see/agree @dmikusa, doesn't make sense to add more routes to the Java example than is unnecessary  once you can configure it at a Kubernetes level and also breaks the consistency across the rest of the Java examples

For the node apps it would be nice to have a `/health` or `/actuator/health` (to match the java pattern) endpoint rather than the `/` endpoint which is meant to serve content. Of course is not going to be something more than a hardcoded  `{status: "UP}` response, but we don't have something to actually check (e.g. a database) so I think it makes sense in that context. 

+1 for adding the kubernetes configuration to the docs.
 
Thank you for your feedback :)

**Comment by dmikusa** [comment]
> For the node apps it would be nice to have a /health or /actuator/health (to match the java pattern) endpoint rather than the / endpoint which is meant to serve content. Of course is not going to be something more than a hardcoded {status: "UP} response, but we don't have something to actually check (e.g. a database) so I think it makes sense in that context.

That sounds good to me. Making the samples more consistent across languages is 👍

**Review by anthonydahanne — APPROVED** [review]

**Comment by mhdawson** [comment]
Adding the single endpoint to the Node.js one to be consistent with the Java example sounds good to me.

**Review by mhdawson — APPROVED** [review]
LGTM

**Review by dmikusa — APPROVED** [review]

**Pull request merged**